conda/activate.py

```diff
@@ -971,19 +971,20 @@ def _hook_preamble(self) -> str:
             "profile.d",
             "conda.csh",
         )
+        conda_exe = context.conda_exe_vars_dict["CONDA_EXE"]
```

Comment:
I've contemplated doing the same before but doing it this way is incorrect IMO. This completely ignores the purpose of deprecating `context.conda_exe` in favor of `context.conda_exe_vars_dict`.

Instead we should be doing something similar to what is done in some of the other `_hook_preamble` implementations, see `conda.activate.PosixActivator._hook_preamble`:

https://github.com/conda/conda/blob/062354f4501f2561945af95b52c81e460c4a03d0/conda/activate.py#L924-L935

Reply:
Okay, I'll look at this. I had no idea why `context.conda_exe` was deprecated, so was just trying to get things working.